addons/binding/org.openhab.binding.tadoac/src/main/java/org/openhab/binding/tadoac/handler/TadoACHandler.java

```diff
@@ -0,0 +1,216 @@
+                    setting.setMode(TadoACMode.DRY);
+                    break;
+                case 3:
+                    setting.setMode(TadoACMode.FAN);
+                    break;
+                default:
+                    logger.error("mode channel out of range");
+            }
+        } else if (channelUID.getId().equals(CHANNEL_TEMP) && command instanceof DecimalType) {
+            float temperature = ((DecimalType) command).floatValue();
+            TadoACTemperature tadotemp = new TadoACTemperature();
+            tadotemp.setCelsius(temperature);
+            setting.setTemperature(tadotemp);
+        } else if (channelUID.getId().equals(CHANNEL_FANSPEED) && command instanceof DecimalType) {
+            int speed = ((DecimalType) command).intValue();
+            switch (speed) {
+                case 1:
+                    setting.setFanSpeed(TadoACFanSpeed.LOW);
+                    break;
+                case 2:
+                    setting.setFanSpeed(TadoACFanSpeed.MIDDLE);
+                    break;
+                case 3:
+                    setting.setFanSpeed(TadoACFanSpeed.HIGH);
+                    break;
+                default:
+                    logger.error("mode channel out of range");
+            }
+        }
+
+        synchronized (this) {
+            if (sendFuture != null) {
+                sendFuture.cancel(false);
+            }
+            sendFuture = scheduler.schedule(() -> sendSetting(), 1, TimeUnit.SECONDS);
+        }
+    }
+
+    private synchronized void sendSetting() {
+        try {
+            String username = (String) getConfig().get("username");
+            String password = (String) getConfig().get("password");
+            int homeid = ((BigDecimal) getConfig().get("homeid")).intValue();
+            int zoneid = ((BigDecimal) getConfig().get("zoneid")).intValue();
+            if (username == null || password == null) {
+                logger.error("login credentials not configured");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+
+            // validate setting
+            if (setting.getPower() == null) {
+                logger.error("No power setting");
+                return;
+
+            }
+            if (setting.getPower() == TadoACPower.ON) {
+                if (setting.getMode() == null) {
+                    logger.error("No mode setting");
+                    return;
+                }
+                if (setting.getFanSpeed() == null) {
+                    logger.error("No fan setting");
+                    return;
+                }
+                if (setting.getMode() == TadoACMode.COOL && setting.getTemperature() == null) {
+                    logger.error("No temperature setting");
+                    return;
+                }
+            }
+            TadoACConnector connector = new TadoACConnector(username, password);
+            connector.setSetting(homeid, zoneid, setting);
```

Comment:
If this call fails there is an IOException which should be caught and then the thing should go into OFFLINE state.